runtime/src/commonMain/kotlin/pbandk/ExtendableMessage.kt

```diff
@@ -0,0 +1,21 @@
+package pbandk
+
+@Suppress("UNCHECKED_CAST")
+interface ExtendableMessage : Message {
+    val extensionFields: pbandk.AtomicReference<Map<Int, Any>>
+    fun <M: Message, T> getExtension(fd: FieldDescriptor<M, T>): T {
+        val value = (extensionFields.get()[fd.number] as? T)
```

Comment:
If the field was already in `extensionFields` we should just return it. The current code will reinsert the value into the map on every read access.